src/crons/nft/nft.cron.service.ts

```diff
@@ -37,6 +41,27 @@ export class NftCronService {
     }, true);
   }
 
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async triggerProcessWhitelistedNfts() {
+    if (!this.apiConfigService.getIsProcessNftsFlagActive()) {
+      return;
+    }
+
+    let lastProcessedTimestamp = await this.cachingService.getCache<number>(CacheInfo.LastProcessedTimestamp.key) ?? Math.floor(Date.now() / 1000);
+    this.logger.log(`Last processed timestamp until now: ${lastProcessedTimestamp} (${new Date(lastProcessedTimestamp * 1000)})`);
+
+    await Locker.lock('Process NFTs whitelisted', async () => {
+      const nfts = await this.nftService.getNfts(new QueryPagination({ from: 0, size: 2500 }), new NftFilter({ isWhitelistedStorage: true, hasUris: true, before: lastProcessedTimestamp }));
+
+      await Promise.all(nfts.map((nft: Nft) => this.nftWorkerService.addProcessNftQueueJob(nft, new ProcessNftSettings({ uploadAsset: true }))));
```

Comment:
I would make the check before triggering the rabbitmq, so that we don't overload rmq with messages